examples/dynamo/README

```diff
@@ -196,6 +196,46 @@ and manipulated:
     cd eg11/
     psyclone -s ./async_script.py helmholtz_solver_alg_mod.x90
 
+Example 12: Code extraction
```

Comment:
Removed the `extract_single_kernel.py` script. Could not really think about the use case for `extract_kernel_multi_invokes.py` so I removed it, too. My idea here was to create multiple micro-benchmarks from one algorithm, however this would be better achieved by extracting nodes from different invokes than the same Kernel.
Renamed "optimisations" to "transformations" throughout and deleted a lot of code from the algorithm file. I also amended the `find_kernel.py` script to write out the name and schedule of Invokes containing the call to the specified Kernel.